asapdiscovery-ml/asapdiscovery/ml/utils.py

```diff
@@ -244,104 +247,101 @@ def build_model(
-            elif pred_readout == "none":
-                pred_readout = None
-            else:
-                raise ValueError(f"Uknown value for -pred_r: {pred_readout}")
-        except AttributeError:
+    # Check and parse pred readout
+    try:
+        pred_readout = (
+            config["pred_r"].lower() if "pred_r" in config else pred_r.lower()
+        )
+        if pred_readout == "pic50":
+            pred_readout = mtenn.model.PIC50Readout()
+        elif pred_readout == "none":
             pred_readout = None
+        else:
+            raise ValueError(f"Uknown value for -pred_r: {pred_readout}")
+    except AttributeError:
+        pred_readout = None
 
-        # Check and parse comb readout
-        try:
-            comb_readout = (
-                config["comb_r"].lower() if "comb_r" in config else comb_r.lower()
-            )
-            if comb_readout == "pic50":
-                comb_readout = mtenn.model.PIC50Readout()
-            elif comb_readout == "none":
-                comb_readout = None
-            else:
-                raise ValueError(f"Uknown value for -comb_r: {comb_readout}")
-        except AttributeError:
+    # Check and parse comb readout
+    try:
+        comb_readout = (
+            config["comb_r"].lower() if "comb_r" in config else comb_r.lower()
+        )
+        if comb_readout == "pic50":
+            comb_readout = mtenn.model.PIC50Readout()
+        elif comb_readout == "none":
             comb_readout = None
+        else:
+            raise ValueError(f"Uknown value for -comb_r: {comb_readout}")
+    except AttributeError:
+        comb_readout = None
 
-        # Use previously built model to construct mtenn.model.Model
-        model = get_model(
-            model=model,
+    # Build initial model object, which will be used later in the get_model call
+    if model_type == "gat":
+        model = build_model_2d(config)
+        get_model = mtenn.conversion_utils.gat.GAT.get_model
+    elif model_type == "schnet":
+        model = build_model_schnet(config)
+        get_model = partial(
+            mtenn.conversion_utils.schnet.SchNet.get_model,
             grouped=grouped,
             strategy=strategy,
             combination=combination,
-            pred_readout=pred_readout,
             comb_readout=comb_readout,
-            fix_device=True,
         )
 
-        def model_call(model, d):
-            return model(d)
-
+    elif model_type == "e3nn":
+        # Load model parameters
+        if (type(e3nn_params) is list) or (type(e3nn_params) is tuple):
+            model_params = e3nn_params
+        elif os.path.isfile(e3nn_params):
+            model_params = pkl.load(open(e3nn_params, "rb"))
+        else:
+            raise ValueError(
+                "Must provide an appropriate value for e3nn_params "
+                f"(received {e3nn_params})"
+            )
+        model = build_model_e3nn(100, *model_params[1:], config)
+        get_model = partial(
```

Comment:
Nice use of `partial`, love it ;)

Reply:
haha thanks, I actually didn't know about this functionality until I saw you use it somewhere, but partially applying functions is one of my favorite things from functional programming, so expect to see a lot more of it :)